pkg/util/kubernetes/kubelet/types_kubelet.go

```diff
@@ -69,11 +69,17 @@ type ContainerProbe struct {
 
 // Status contains fields for unmarshalling a Pod.Status
 type Status struct {
-	Phase      string            `json:"phase,omitempty"`
-	HostIP     string            `json:"hostIP,omitempty"`
-	PodIP      string            `json:"podIP,omitempty"`
-	Containers []ContainerStatus `json:"containerStatuses,omitempty"`
-	Conditions []Conditions      `json:"conditions,omitempty"`
+	Phase          string            `json:"phase,omitempty"`
+	HostIP         string            `json:"hostIP,omitempty"`
+	PodIP          string            `json:"podIP,omitempty"`
+	Containers     []ContainerStatus `json:"containerStatuses,omitempty"`
+	InitContainers []ContainerStatus `json:"initContainerStatuses,omitempty"`
+	Conditions     []Conditions      `json:"conditions,omitempty"`
+}
+
+// AllContainers returns the list of init and regular containers
+func (s *Status) AllContainers() []ContainerStatus {
+	return append(s.InitContainers, s.Containers...)
```

Comment:
we could probably lazy load this array ?

Reply:
Hesitated to do so, because I was unsure the object was read-only, but should be safe, done